internal/namespaces/rdb/v1/custom_instance.go

```diff
@@ -163,6 +163,184 @@ func instanceUpgradeBuilder(c *core.Command) *core.Command {
+				Deprecated: false,
+				Positional: false,
+			},
+			{
+				Name:       "name",
+				Short:      `Name of the instance`,
+				Required:   false,
+				Deprecated: false,
+				Positional: false,
+			},
+			{
+				Name:       "instance-id",
+				Short:      `UUID of the instance to update`,
+				Required:   true,
+				Deprecated: false,
+				Positional: true,
+			},
+			{
+				Name:       "tags.{index}",
+				Short:      `Tags of a given instance`,
+				Required:   false,
+				Deprecated: false,
+				Positional: false,
+			},
+			{
+				Name:       "logs-policy.max-age-retention",
+				Short:      `Max age (in day) of remote logs to keep on the database instance`,
+				Required:   false,
+				Deprecated: false,
+				Positional: false,
+			},
+			{
+				Name:       "logs-policy.total-disk-retention",
+				Short:      `Max disk size of remote logs to keep on the database instance`,
+				Required:   false,
+				Deprecated: false,
+				Positional: false,
+			},
+			{
+				Name:       "backup-same-region",
+				Short:      `Store logical backups in the same region as the database instance`,
+				Required:   false,
+				Deprecated: false,
+				Positional: false,
+			},
+			{
+				Name:       "settings.{index}.name",
+				Short:      `Setting name of a given instance`,
+				Required:   false,
+				Deprecated: false,
+				Positional: false,
+			},
+			{
+				Name:       "settings.{index}.value",
+				Short:      `Setting value of a given instance`,
+				Required:   false,
+				Deprecated: false,
+				Positional: false,
+			},
+			core.RegionArgSpec(scw.RegionFrPar, scw.RegionNlAms, scw.RegionPlWaw),
+		},
+		Run: func(ctx context.Context, args interface{}) (i interface{}, e error) {
+			customRequest := args.(*rdbUpdateInstanceRequestCustom)
+
+			updateInstanceRequest := customRequest.UpdateInstanceRequest
+
+			client := core.ExtractClient(ctx)
+			api := rdb.NewAPI(client)
+
+			getResp, err := api.GetInstance(&rdb.GetInstanceRequest{
+				Region:     customRequest.Region,
+				InstanceID: customRequest.InstanceID,
+			})
+			if err != nil {
+				return nil, err
+			}
+
+			if customRequest.Settings != nil {
+				settings := getResp.Settings
+				changes := customRequest.Settings
+
+				for _, change := range changes {
+					matched := false
+					for _, setting := range settings {
+						if change.Name == setting.Name {
+							setting.Value = change.Value
+							matched = true
+							break
+						}
+					}
+					if !matched {
+						settings = append(settings, change)
+					}
+				}
+
+				_, err = api.SetInstanceSettings(&rdb.SetInstanceSettingsRequest{
+					Region:     updateInstanceRequest.Region,
+					InstanceID: updateInstanceRequest.InstanceID,
+					Settings:   settings,
+				})
+				if err != nil {
+					return nil, err
+				}
+			}
+
+			updateInstanceResponse, err := api.UpdateInstance(updateInstanceRequest)
+			if err != nil {
+				return nil, err
+			}
+
+			resp, err := api.GetInstance(&rdb.GetInstanceRequest{
```

Comment:
No need to fetch the instance after updating it, `UpdateInstance` returns the updated instance.